### PR TITLE
156 refactor sending messages

### PR DESCRIPTION
**Pull request by summer-cook**
# Story
brings back the ability to send messages, and adds the quotedWareID to the request so it can be passed in the post message request as well as in the posting method since it needed to be used when adding initial attachments

Ref #156 

# Expected Behavior Before Changes
messages could not be sent from the webstore, and were sent with the incorrect param of quote_group_id vs the ID of the quoted_ware. 

# Expected Behavior After Changes
- [ ] messages can be sent from the webstore to the correct quote_group/quoted_ware combo
- [ ] attachments can be sent when creating a request with the correct quote_group/quoted_ware combo

### CODE REVIEW

**Comment by summer-cook** [comment]
@alishaevn I'm going to change this ticket back to a 2, because it look a lot more than just updating the quoted_ware_ids array to use the quotedWareID to bring back all of the message & attachment sending functionality, since it needed to be used when creating requests as well

also, here is the related PR uncommenting everything in the component library: https://github.com/scientist-softserv/webstore-component-library/pull/161/files

**Comment by summer-cook** [comment]
also, @alishaevn should I change the variable of `quotedWareID` to either `serviceID` or just `wareID` ?

I felt like it might need to be distinguished since there are SO many ids

File: pages/requests/[id].js [hunk 1]
```diff
@@ -67,27 +66,23 @@ const Request = () => {
...
   return (
     <div className='container'>
       <StatusBar statusArray={STATUS_ARRAY} apiRequestStatus={request.status.text} addClass='mt-4' />
       <div className='row mb-4'>
         <div className='col-sm-4 col-md-3 mt-2 mt-sm-4 order-1 order-sm-0'>
-          {/* TODO(@summercook): 
-          - add back in the handleSendingMessagesOrFiles={handleSendingMessagesOrFiles} prop
-            to ActionsGroup once posting messages/attachments has been refactored. */}
-          <ActionsGroup initialFiles={files}/>
+          <ActionsGroup initialFiles={files} handleSendingMessagesOrFiles={handleSendingMessagesOrFiles}/>
```
Comment (alishaevn) [inline pages/requests/[id].js:85]:
```suggestion
          <ActionsGroup
            initialFiles={files}
            handleSendingMessagesOrFiles={handleSendingMessagesOrFiles}
          />
```

Comment (alishaevn) [reply pages/requests/[id].js:85]: since we know there will be more actions in the future.

File: utils/api/base.js [hunk 2]
```diff
@@ -18,12 +18,14 @@ export const fetcher = (url, token) => {
 export const posting = async (url, data, token) => {
   try {
     const response = await api.post(url, data, { headers: defaultHeaders(token) })
-
-    if (response.data.id) {
+    let quotedWareID = response.data.quoted_ware_refs?.[0].id
+    let requestID = response.data.id
+    if (requestID && quotedWareID) {
```
Comment (alishaevn) [inline utils/api/base.js:23]: we're using `posting` for other functions too, and potentially more as the app expands. does the `quotedWareID` _always_ come back in the data object??

File: utils/api/configurations.js [hunk 3]
```diff
@@ -57,6 +57,7 @@ export const configureRequests = ({ data, path }) => {
       title: `${request.identifier}: ${request.name}`,
       updatedAt: normalizeDate(request.updated_at),
       uuid: request.uuid,
+      quotedWareID: request.quoted_ware_refs?.[0].id
```
Comment (alishaevn) [inline utils/api/configurations.js:60]:
```suggestion
      quotedWareID: request.quoted_ware_refs?.[0].id,
```

File: utils/api/base.js [hunk 4]
```diff
@@ -18,12 +18,14 @@ export const fetcher = (url, token) => {
...
       return {
         success: true,
         error: false,
-        requestID: response.data.id,
+        requestID,
+        quotedWareID
```
Comment (alishaevn) [inline utils/api/base.js:28]:
```suggestion
        quotedWareID,
```

**Review by alishaevn — CHANGES_REQUESTED** [review]

**Comment by alishaevn** [comment]
> also, @alishaevn should I change the variable of `quotedWareID` to either `serviceID` or just `wareID` ?
> 
> I felt like it might need to be distinguished since there are SO many ids

I'd keep it at `quotedWareID` because it is not a ware/service ID. 🙂

**Review by alishaevn — APPROVED** [review]

**Pull request merged**